src/FshConversionProvider.ts

```diff
@@ -0,0 +1,255 @@
+
+  return fhirVersion;
+}
+
+function getDependenciesFromSushiConfig(
+  config: SushiConfiguration
+): { packageId: string; version: string }[] {
+  //try to get dependencies: more or less doing SUSHI's importConfiguration.parseDependencies
+  let parsedDependencies: { packageId: string; version: string }[] = [];
+  if (config.dependencies) {
+    parsedDependencies = Object.entries(config.dependencies).map(
+      ([packageId, versionOrDetails]) => {
+        if (typeof versionOrDetails === 'string' || typeof versionOrDetails === 'number') {
+          return { packageId, version: `${versionOrDetails}` };
+        } else if (versionOrDetails == null) {
+          return { packageId, version: undefined };
+        } else {
+          return {
+            packageId,
+            version: versionOrDetails.version ? `${versionOrDetails.version}` : undefined
+          };
+        }
+      }
+    );
+  }
+  return parsedDependencies;
+}
+
+export async function findMatchingFSHResourcesForProject(configFileUri: Uri): Promise<string[]> {
+  const fshResources: string[] = [];
+  const configdir = dirname(configFileUri.fsPath) + path.sep;
+  const files: Uri[] = await workspace.findFiles('**/*.{fsh}');
+
+  for (const file of files) {
+    if (file.fsPath.startsWith(configdir)) {
+      //Found FSH resource belonging to project.
+      const contents = await workspace.fs.readFile(file);
+      const decoder = new TextDecoder();
+      const decodedContents = decoder.decode(contents);
+      fshResources.push(decodedContents);
+    }
+  }
+
+  return fshResources;
+}
+
+export async function findMatchingJsonResourcesForProject(configFileUri: Uri): Promise<string[]> {
+  const jsonResources: string[] = [];
+  const configdir = dirname(configFileUri.fsPath) + path.sep;
+  const files: Uri[] = await workspace.findFiles('**/*.{json}');
+
+  for (const file of files) {
+    if (file.fsPath.startsWith(configdir)) {
+      //Found FSH resource belonging to project.
+      const contents = await workspace.fs.readFile(file);
+      const decoder = new TextDecoder();
+      const decodedContents = decoder.decode(contents);
+      jsonResources.push(decodedContents);
+    }
+  }
+
+  return jsonResources;
+}
+
+export async function findNamesInFSHResource(fshResource: Uri): Promise<string[]> {
+  const ids: string[] = [];
+
+  const fshContent = await workspace.fs.readFile(fshResource);
+  const decoder = new TextDecoder();
+  const fshString = decoder.decode(fshContent);
+
+  const lines = fshString.split('\n');
+  for (const line of lines) {
+    const match = line.match(
+      /(Instance|Profile|Extension|ValueSet|CodeSystem|Logical|Resource)\s*:\s*(\S*)/
+    );
+    if (match) {
+      ids.push(match[2]);
+    }
+  }
+
+  return ids;
+}
+
+export function findFSHResourceInResult(fshResult: gofshClient.fshMap, resourceId: string): string {
+  let resourceContent = fshResult.aliases + '\n\n';
```

Comment:
This is the safest approach -- as including too many aliases does not hurt anything. But... it does result in outputting a bunch of aliases that are not relevant to the FSH code that was produced. For the project I tested this on, converting FHIR to FSH on one of my JSON files resulted in a preview FSH file with 43 aliases and 9 lines of FSH (which didn't actually use _any_ of those aliases).

I think it would be friendlier to include only the aliases that are relevant. I felt bad asking you for another change, so I went ahead and tried implementing it myself. This seems to work well for me:
```ts
export function findFSHResourceInResult(fshResult: gofshClient.fshMap, resourceId: string): string {
  const resultKeys: string[] = [
    'mappings',
    'profiles',
    'extensions',
    'logicals',
    'resources',
    'codeSystems',
    'valueSets',
    'instances'
  ];

  let resourceContent = '';
  for (const [key, resourceMap] of Object.entries(fshResult)) {
    if (resultKeys.includes(key)) {
      (resourceMap as gofshClient.ResourceMap).forEach((value, id) => {
        if (id === resourceId) {
          resourceContent += value;
        }
      });
    }
  }

  const usedAliasLines = fshResult.aliases.split(/\r?\n/).filter(line => {
    const match = line.match(/^Alias:\s+([^\s]+)\s+/);
    return match && resourceContent.indexOf(match[1]) >= 0;
  });
  if (usedAliasLines.length) {
    resourceContent = usedAliasLines.join(os.EOL) + `${os.EOL}${os.EOL}` + resourceContent;
  }

  return resourceContent;
}
```

Reply:
Thanks Chris, Looks good. I did test it in several conversions, and it works fine. This is indeed better than just adding all aliases.